### PR TITLE
Failure resistance

### PR DESCRIPTION
**Pull request by halink0803**
- [x] API for the analytic to push notification. This API has following params: exchange, action, token pair, from, to, isWarning, msg. Exchange is the name of an exchange that analytic wants to notify about. Msg is the message for it. from, to is timestamp of that the notification covers, they are in milisec.

- [x] Dont allow analytic to disable/enable exchange anymore

- [x] API to get all of current notification that analytic has pushed (including all exchanges)

### CODE REVIEW

**Comment by halink0803** [comment]
[support new failure resistance features](https://trello.com/c/mF54P4WZ/193-support-new-failure-resistance-features)

File: data/storage/bolt.go [hunk 1]
```diff
@@ -1027,3 +1029,61 @@ func (self *BoltStorage) UpdateExchangeStatus(data common.ExchangesStatus) error
 	})
 	return err
 }
+
+func (self *BoltStorage) UpdateExchangeNotification(
+	exchange, action, token string, fromTime, toTime uint64, isWarning bool, msg string) error {
```
Comment (tranvictor) [inline data/storage/bolt.go]: lets not encode all notifications into a json string anymore. we should have different bucket for different exchange, each bucket has different key for each pair.

File: data/storage/bolt.go [hunk 2]
```diff
@@ -1027,3 +1029,61 @@ func (self *BoltStorage) UpdateExchangeStatus(data common.ExchangesStatus) error
...
+			IsWarning: isWarning,
+			Message:   msg,
+		}
+		exchangeAction[action] = exchangeToken
+		notifications[exchange] = exchangeAction
+
+		// remove old data
+		if k != nil {
+			b.Delete(k)
+		}
```
Comment (tranvictor) [inline data/storage/bolt.go]: Also Delete here is redundant. Put can also overwrite data as i remember.

Comment (halink0803) [reply data/storage/bolt.go]: As I chose the key by timepoint then different timepoint will have different key, then it needs to be deleted. However, I will change as you recommended above using exchange for bucket name and action_token for key then we don't need to delete anymore.

File: data/storage/bolt.go (hunk 1 above)
Comment (halink0803) [reply data/storage/bolt.go]: updated

**Pull request merged**